File: chapter4/java/src/main/java/com/dotComIt/learnWith/config/DatabaseConfig.java
package com.dotComIt.learnWith.config;

import java.sql.Connection;
import java.sql.DriverManager;
import com.microsoft.sqlserver.jdbc.SQLServerDriver;

public class DatabaseConfig {
    String host = "jdbc:sqlserver://localhost";
    String port = "1433";
    String database = "LearnWithApp";
    String username = "LearnWithUser";
    String password = "password";
    Connection connection = null;

    public Connection openConnection(){
        if(connection != null){
            return connection;
        }
        String connectionString = host + ":" + port + ";"
                + "database=" + database + ";"
                + "user=" + username + ";"
                + "password=" + password + ";";
        try {
            Class.forName("com.microsoft.sqlserver.jdbc.SQLServerDriver");
            connection = DriverManager.getConnection(connectionString);
        }
        catch (Exception e) {
            System.out.println("Exception");
            e.printStackTrace();
        }
        return connection;
    }

    public void closeConnection(){
        try { connection.close(); } catch(Exception e) {}
    }

}

File: chapter4/php/com/dotComIt/learnWith/config/DatabaseConfig.php
<?php
    class DatabaseConfig
    {
        private $host = "localhost";
        private $port = "1433";
        private $db_name = "LearnWithApp";
        private $username = "LearnWithUser";
        private $password = "password";

        public $conn;

        public function openConnection(){
            if($this->conn){
                return $this->conn;
            };

            $serverName = "tcp:$this->host,$this->port";
            $connectionString = "sqlsrv:server=$serverName; Database=$this->db_name";
            try{
                $this->conn = new PDO( $connectionString, $this->username, $this->password);
                if($this->conn == false){
                    die(FormatErrors(sqlsrv_errors()));
                }
            }catch(Exception $exception){
                echo "Connection error: " . $exception->getMessage();
            }

            return $this->conn;

        }

        public function closeConnection(){
            $this->conn = null;
        }


    }

?>